independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/ConfigService.java

```diff
@@ -0,0 +1,194 @@
+package io.quarkus.analytics;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+
+import io.quarkus.analytics.config.FileLocations;
+import io.quarkus.analytics.config.GroupIdFilter;
+import io.quarkus.analytics.dto.config.AnalyticsRemoteConfig;
+import io.quarkus.analytics.dto.config.LocalConfig;
+import io.quarkus.analytics.dto.config.NoopRemoteConfig;
+import io.quarkus.analytics.dto.config.RemoteConfig;
+import io.quarkus.analytics.rest.ConfigClient;
+import io.quarkus.analytics.util.FileUtils;
+import io.quarkus.devtools.messagewriter.MessageWriter;
+
+/**
+ * Decided the build analytics behaviour. Retrieves, stores and provides the configuration.
+ */
+public class ConfigService {
+    public static final String QUARKUS_ANALYTICS_DISABLED_LOCAL_PROP = "quarkus.analytics.disabled";
+    public static final String QUARKUS_ANALYTICS_PROMPT_TIMEOUT = "quarkus.analytics.prompt.timeout";
+    private static final int DEFAULT_REFRESH_HOURS = 12;
+    private static final String NEW_LINE = System.lineSeparator();
+    static final String ACCEPTANCE_PROMPT = NEW_LINE +
+            "*****************************************************************************************************" + NEW_LINE
+            + " Do you agree to send anonymous usage data to the Quarkus community to help improve Quarkus? (y/n) " + NEW_LINE;
```

Comment:
Is it obvious from this question that the usage data will not be gathered from the built application? I think it is, but maybe I'm too close to the problem to see it clearly.

Reply:
I agree with you. What about:
`Do you agree to send build time anonymous usage data to the Quarkus community to help improve Quarkus? (y/n)`

Reply:
Would 'development time' be more obvious to people, whilst still conveying the right message?

Reply:
It could, but we use `build time analytics` everywhere else. It might confuse ppl.

Reply:
Ah OK. Agreed, we should keep the terms consistent.

I think flipping the order reads better: 

`Do you agree to send anonymous build time usage data to the Quarkus community to help improve Quarkus? (y/n)`

I don't know enough about English grammar to say for sure. But I think it now sounds like the ordering is correct based on this weird rule: https://dictionary.cambridge.org/grammar/british-grammar/adjectives-order.

@rolfedh can you review this prompt text too please? In particular, should we hyphenate "build time", in this context, to remove ambiguity?  But any other general review would be welcome too!

Reply:
Updated the line